src/pages/sa-ord-dtd/forms/DocumentTypeDefaultForm.js

```diff
@@ -0,0 +1,182 @@
+import { Checkbox, FormControlLabel, Grid } from '@mui/material'
+import { useContext, useEffect } from 'react'
+import * as yup from 'yup'
+import FormShell from 'src/components/Shared/FormShell'
+import toast from 'react-hot-toast'
+import { RequestsContext } from 'src/providers/RequestsContext'
+import { useInvalidate } from 'src/hooks/resource'
+import { ResourceIds } from 'src/resources/ResourceIds'
+import { useForm } from 'src/hooks/form'
+import { Grow } from 'src/components/Shared/Layouts/Grow'
+import { VertLayout } from 'src/components/Shared/Layouts/VertLayout'
+import { ControlContext } from 'src/providers/ControlContext'
+import { SaleRepository } from 'src/repositories/SaleRepository'
+import CustomNumberField from 'src/components/Inputs/CustomNumberField'
+import ResourceComboBox from 'src/components/Shared/ResourceComboBox'
+import { SystemRepository } from 'src/repositories/SystemRepository'
+import { DataSets } from 'src/resources/DataSets'
+
+export default function DocumentTypeDefaultForm({ labels, maxAccess, dtId }) {
+  const { getRequest, postRequest } = useContext(RequestsContext)
+  const { platformLabels } = useContext(ControlContext)
+
+  const invalidate = useInvalidate({
+    endpointId: SaleRepository.DocumentTypeDefault.qry
+  })
+
+  const { formik } = useForm({
+    initialValues: {
+      dtId: dtId || null,
+      recordId: '',
+      spId: '',
+      commitItems: false,
+      disableSKULookup: false,
+      allocateBy: ''
+    },
+    enableReinitialize: true,
+    validateOnChange: true,
+    validationSchema: yup.object({
+      dtId: yup.string().required()
+    }),
+
+    onSubmit: async obj => {
+      try {
+        const response = await postRequest({
+          extension: SaleRepository.DocumentTypeDefault.set,
+          record: JSON.stringify(obj)
+        })
+
+        if (!formik.values.recordId) {
+          toast.success(platformLabels.Added)
+          formik.setFieldValue('recordId', formik.values.dtId)
+        } else toast.success(platformLabels.Edited)
+
+        invalidate()
+      } catch (error) {}
+    }
+  })
+  console.log(formik.values)
+  const editMode = !!formik.values.recordId
+  useEffect(() => {
+    ;(async function () {
+      try {
+        if (dtId) {
+          const res = await getRequest({
+            extension: SaleRepository.DocumentTypeDefault.get,
+            parameters: `_dtId=${dtId}`
+          })
+
+          formik.setValues({ ...res.record, recordId: dtId })
+        }
+      } catch (error) {}
+    })()
+  }, [])
+
+  return (
+    <FormShell resourceId={ResourceIds.DocumentTypeDefault} form={formik} maxAccess={maxAccess} editMode={editMode}>
+      <VertLayout>
+        <Grow>
+          <Grid container spacing={4}>
+            <Grid item xs={12}>
+              <ResourceComboBox
```

Comment:
if this field is required, add required to it and if not, remove it form validationSchema